File: gateway/src/main/java/com/beat/gateway/guest/internal/store/GuestSession.java
package com.beat.gateway.guest.internal.store;

import org.springframework.data.annotation.Id;
import org.springframework.data.redis.core.RedisHash;

@RedisHash(value = "guestSession", timeToLive = 1800)
public class GuestSession {

	@Id
	private String tokenHash;

	private Long userId;

	protected GuestSession() {
	}

	private GuestSession(String tokenHash, Long userId) {
		this.tokenHash = tokenHash;
		this.userId = userId;
	}

	public static GuestSession of(String tokenHash, Long userId) {
		return new GuestSession(tokenHash, userId);
	}

	public Long getUserId() {
		return userId;
	}
}

File: gateway/src/main/kotlin/com/beat/gateway/guest/internal/store/GuestSession.kt
package com.beat.gateway.guest.internal.store

import org.springframework.data.annotation.Id
import org.springframework.data.redis.core.RedisHash

@RedisHash(value = "guestSession", timeToLive = 1800)
class GuestSession(
    @Id
    val tokenHash: String,
    val userId: Long,
)
